scripts/c_header_guards.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+"""
+Script to check/fix header guards in C header files.
+Converts guards to the format
+    TF_PSA_CRYPTO_<rel_path>_<filename>_H
+where:
+- 'TF_PSA_CRYPTO_' is the common hardcoded prefix;
+- 'rel_path' is the path used in C files to include such header file.
```

Comment:
That's not my understanding of the rules we agreed on. I think the rule were:
- first candidate is `<rel_path>_<filename>_H`;
- if this first candidate starts with `MBEDTLS_` or `PSA_` or `TF_PSA_CRYPTO_` then keep it;
- else add a `TF_PSA_CRYPTO_` prefix.

See https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/460#issuecomment-3311375010

This results in shorter names for some files by not adding a `TF_PSA_CRYPTO_` prefix when `<rel_path>_<filename>_H` naturally lands in one of our 3 namespaces.